src/UserRepository/PdoDatabase.php

```diff
@@ -41,14 +42,19 @@ public function __construct(PDO $pdo, array $config)
     public function authenticate(string $credential, string $password = null) : ?UserInterface
     {
         $sql = sprintf(
-            "SELECT %s FROM %s WHERE %s = :username",
+            "SELECT %s FROM %s WHERE %s = :identity",
             $this->config['field']['password'],
             $this->config['table'],
-            $this->config['field']['username']
+            $this->config['field']['identity']
         );
 
         $stmt = $this->pdo->prepare($sql);
-        $stmt->bindParam(':username', $credential);
+        if (false === $stmt) {
+            throw new Exception\RuntimeException(
+                "Error during the user authentication, please check the configuration"
```

Comment:
I'd change this to "An error occurred when preparing to fetch user details from the repository; please verify your configuration." This will give more information to the developer, without giving security details out should the exception message leak to a consumer.

Reply:
@weierophinney good catch, thanks!